### PR TITLE
Allow users to not pass timestamps into SF2 function

### PR DESCRIPTION
**Pull request by drewoldag**
The actual logic here is relatively small (and so hopefully safe). Most of this PR is adding unit tests.

### CODE REVIEW

File: src/lsstseries/analysis/structurefunction2.py [hunk 1]
```diff
@@ -69,6 +70,13 @@ def calc_sf2(
 
             # Mask on band
             times = np.array(time)[band_mask]
+
+            # if times are all `None`, we assume equidistant times between
+            # measurements. To do so, we'll create an array of integers
+            # from 0 to N where N is the number of flux values for this band.
+            if np.all(np.equal(times, None)):
```
Comment (jeremykubica) [inline src/lsstseries/analysis/structurefunction2.py:77]: Small comment and it is fine to leave it as-is: Alternatively (or in addition) you could check if `times` itself is None. This means that the user wouldn't need to allocate an array of Nones.

Comment (drewoldag) [reply src/lsstseries/analysis/structurefunction2.py:77]: I was a little on the fence about supporting that use case, based only the existing docstring. But I do think it's a good idea. So I'll work in the changes necessary to support that.

File: tests/lsstseries_tests/test_analysis.py [hunk 2]
```diff
@@ -83,3 +133,59 @@ def test_dt_bins():
...
+        combine=False,
+        method="size",
+        sthresh=100,
+    )
+
+    assert res["dt"][0] == pytest.approx(3.705, rel=0.001)
+    assert res["sf2"][0] == pytest.approx(0.005365, rel=0.001)
+
+
+def test_sf2_base_case_time_as_none_array():
```
Comment (drewoldag) [inline tests/lsstseries_tests/test_analysis.py:166]: A little extra code coverage, but unrelated to this work.

**Review by dougbrn — APPROVED** [review]
This looks good to me, especially with support for times=None now

**Pull request merged**